manage_projects.rb

```diff
@@ -111,8 +111,28 @@ def add_project
   attributes[:name] = gets.chomp
   print "Host (aws or azure): "
   attributes[:host] = gets.chomp.downcase
-  print "Start date (YYYY-MM-DD): "
-  attributes[:start_date] = gets.chomp
+  valid_date = false
+  while !valid_date
+    print "Start date (YYYY-MM-DD): "
+    valid_date = Date.parse(gets.chomp) rescue false
```

Comment:
It's best not to be so general with `rescue` calls like this. It's better to rescue a specific kind of error (in this case, `ArgumentError`), otherwise you're rescuing  `StandardError` and _all_ of it's subclasses. Unfortunately you can't rescue a specific error inline, so you'll have to do a `begin`-`rescue` block:
```
valid_date = begin
  Date.parse(gets.chomp)
rescue ArgumentError
  false
end
```

As a side note, this is definitely something I should have been pointing out with _every_ usage of inline rescue over the last couple weeks, but the importance of it has been slipping my mind. I'll be logging it as an issue.